lib/Db/PimcoreExtensionsTrait.php

```diff
@@ -341,8 +343,13 @@ public function fetchPairs($sql, array $params = [], $types = [])
     public function insertOrUpdate($table, array $data)
     {
         // get the field name of the primary key
-        $tableDetails = $this->getSchemaManager()->listTableDetails($table);
-        $fieldsPrimaryKey = (array)$tableDetails->getPrimaryKeyColumns();
+        if(isset(self::$tablePrimaryKeyCache[$table])) {
+            $fieldsPrimaryKey = self::$tablePrimaryKeyCache[$table];
+        } else {
+            $tableDetails = $this->getSchemaManager()->listTableDetails($table);
+            $fieldsPrimaryKey = (array) $tableDetails->getPrimaryKeyColumns();
+            self::$tablePrimaryKeyCache[$table] = $fieldsPrimaryKey;
+        }
```

Comment:
This could be written as follows - not sure if it's more readable, though:

```suggestion
        $fieldsPrimaryKey = self::$tablePrimaryKeyCache[$table] ??= (array) $this->getSchemaManager()->listTableDetails($table)->getPrimaryKeyColumns();
```

Reply:
Yep, makes sense, thanks 😊